sirepo/package_data/static/js/srw.js

```diff
@@ -2533,6 +2534,28 @@ SIREPO.app.directive('simulationStatusPanel', function(appState, beamlineService
                 return $scope.model == 'coherentModesAnimation';
             }
 
+            function coherentModesCoresError(cores, minCores) {
+                if (cores < minCores) {
+                    throw new Error(`${cores} cores found, should be > 3 for coherent modes`);
```

Comment:
Where does this show up?

Reply:
Do you mean where does the error message show up? In the page header.

Reply:
I think this is still the wrong way about it. Since this is an error in configuration, it really should originate from the server and come back as a UserAlert. This way the server protects itself. If there are other users of the API (e.g. tests or bluesky), the server will not produce a reasonable error message.

Reply:
Yes, I agree it should be on the server and it should throw an exception which would automatically be displayed in the report window. The problem is we don't want it to throw the exception when the code is generated, only when it is run and we can't find the appropriate hook for this on the server because pkcli.srw doesn't use run_background(). Where is the hook in the server to do this check?

Reply:
```diff
--- a/sirepo/template/srw.py
+++ b/sirepo/template/srw.py
@@ -1052,6 +1052,8 @@ def write_parameters(data, run_dir, is_parallel):
         p = ""
         _export_rsopt_config(data)
     else:
+        if data.report == "coherentModesAnimation":
+            raise sirepo.util.UserAlert("ruh roh!")
         p = _trim(_generate_parameters_file(data, run_dir=run_dir))
     pkio.write_text(
         run_dir.join(template_common.PARAMETERS_PYTHON_FILE),
```

Reply:
The error from UserAlert doesn't show when in_slurm == True.
eg.
```
Simulation Error: no reply
Elapsed time: 00:00:05
```

 It will only show when not in vagrant/slurm.

Reply:
That's a bug. Create an issue. UserAlerts should cascade.